### DISCUSSION TITLE
maintenance status integration (solved)

### DISCUSSION THREAD

**Discussion by Timongcraft** [PetyXbron/minecraft-bot#7] [Ideas]
Can you impement the https://www.spigotmc.org/resources/maintenance-bungee-and-spigot-support.40699 plugin that if this is enabled the status is offline or do have I to ask it in `minecraft-server-util`?

**Comment by PetyXbron** [comment]
> Can you impement the https://www.spigotmc.org/resources/maintenance-bungee-and-spigot-support.40699 plugin that if this is enabled the status is offline or do have I to ask it in `minecraft-server-util`?

@Timong-craft you should ask in `minecraft-server-util`. It would be a lot of easier for development.
The only way I can think of is to check the MOTD if it contains the word "maintence".

**Comment by Timongcraft** [comment]
Ok, I will

**Comment by Timongcraft** [comment]
Hello again, they said that the plugin only changes the MOTD and it can be set to anything and it would be a niche status and not worth it implementing.
So, can you implement that when the server MOTD contains the word "maintenance" it sais the server is in maintenance?

**Reply by PetyXbron** [reply]
Okay. I will try to update.
Please have patience when waiting for new release. I will try to start working today, but it may happen that I will release the update in a few days.

**Comment by Timongcraft** [comment]
All right, but don't stress yourself

**Comment by Timongcraft** [comment]
And thanks

**Comment by PetyXbron** [comment]
I published [a commit](https://github.com/PetyXbron/minecraft-bot/commit/c3e5fcaa083c84d867dbc03b099e366af08ada9f) with a version 1.5.8. You can download it and use it for really basic maintenance mode use.

I just replaced online text to maintenance/online based on if motd contains "maintenance" (not caps sensitive) in the statusCH function and in the status command (slash + normal).
I would like to add a help command and a Little bit more breath of the new maintenance mode.
If you have any more suggestions about maintenance mode please let me know here asap.

**Comment by Timongcraft** [comment]
This is awesome, but I think you can't change it in status or?

**Comment by PetyXbron** [comment]
Oh yeah... I didn't think about this function. Sounds cool. I could implement it in the new update. Now I am probably thinking about adding a new variable in config which will be after the "status" and it will exist some another, like "maintenanceStatus".

And I want to add settings to:
- Choice if you want to even detect "maintenance" in motd and if so, you should have a choice to choose the word (another language servers will probably not have "maintenance").

**Comment by Timongcraft** [comment]
That sounds nice and thanks again

**Comment by Timongcraft** [comment]
And may add an option to disable the commands or slashcommands

**Reply by PetyXbron** [reply]
> And may add an option to disable the commands or slashcommands

@Timong-craft you can disable normal commands by setting the `enable` variable to false in each command file.
Classic command `ip` example: [here](https://github.com/PetyXbron/minecraft-bot/blob/main/commands/ip.js#L8)
I will add slash-commands disabling too in the next version. Maybe I should move these settings of enabling certain commands to config.js for better management.

**Reply by Timongcraft** [reply]
Ok
